File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
function solution(survey, choices) {
    var result = {};
    choices.forEach((choice, idx) => {
        var type = '';
        var score = 0;
        if(choice < 4) {
            type = survey[idx][0];
            score = 4 - choice;
        }
        else if(choice > 4) {
            type = survey[idx][1];
            score = choice - 4;
        }
        result[type] ? result[type] += score : result[type] = score;
    })
    console.log(result)
     let answer = (result['R']|0) < (result['T']|0) ? 'T' : 'R';
    answer += (result['C']|0) < (result['F']|0) ? 'F' : 'C';
    answer += (result['J']|0) < (result['M']|0) ? 'M' : 'J';
    answer += (result['A']|0) < (result['N']|0) ? 'N' : 'A';
    return answer;
}

File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "RCJA";
        int[] sum = new int[26];
        int[] score = {3,2,1,0,1,2,3};
        Arrays.fill(sum, 0);
        
        for(int i=0; i<survey.length; i++){
            if(choices[i]<4){
                sum[survey[i].charAt(0)-'A'] += score[choices[i]-1];
            }else if(choices[i]>4){
                sum[survey[i].charAt(1)-'A'] += score[choices[i]-1];
            }
        }
        
        if(sum['R'-'A'] < sum['T'-'A'])  answer = answer.replace("R","T");
        if(sum['C'-'A'] < sum['F'-'A'])  answer = answer.replace("C","F");
        if(sum['J'-'A'] < sum['M'-'A'])  answer = answer.replace("J","M");
        if(sum['A'-'A'] < sum['N'-'A'])  answer = answer.replace("A","N");
        
        return answer;
    }
}